base/doc/ltnews38.tex

```diff
@@ -278,67 +278,149 @@ \section{Hooks, sockets and plugs}
+And the \pkg{smart-ref} package supports the storage of a larger
+collection of counter values so that, for example, a cross-reference
+can refer to the relevant chapter together with an equation tag.
+The \pkg{cleveref} package stores (by means of a second, internal
+\enquote{logical label}) extra information such as the name of the counter.
+The \pkg{hyperref} package adds the \cs{autoref} command, which
+tries to retrieve the name of a counter from the  \emph{logical name}
+used for a link target.
+The \pkg{tikzmarks} library records information about
+ \emph{labelled positions} on the page when using \pkg{tikz}.
+Finally, the \pkg{zref} package implements many related ideas, including a general
+idea of properties and lists of properties, with methods to record, and
+subsequently access, the value of any declared property.
+
+
+%%FMi some rewrite:
+Starting with this release, the \LaTeX{} kernel now offers the handling of general document
+properties as a core functionality with standard interfaces.
+%
+This is based on concepts introduced by the \pkg{zref}
+package but with some differences in detail, particularly
+in the implementation.
+It supports the declaration of new properties, and the recording of
+the values of any list of properties. These values are retrieved expandably.
+
+To set up a new property that is the current chapter
+number, for example, here is the declaration to use.
 \begin{verbatim}
-\NewProperty{chapter}{now}{?}{\thechapter}
+  \NewProperty{chapter}{now}{?}{\thechapter}
 \end{verbatim}
-The second argument means that the property is recorded immediately 
-(\enquote{now}) and not at \cs{shipout}. The third argument sets a default if a 
-label is unknown. It is then possible to record a list of properties with 
-\cs{RecordProperties}: 
+The second argument means that the property value will be recorded
+immediately (\enquote{now}), and not \enquote{during the next \cs{shipout}}.
+The third argument sets a default
+to be used when, for example, an unknown label is supplied.
+%%FMi added
+The final argument contains the code that produces the current
+property value when recording the value is requested.
+
+Then, to record the value of this new property,
+together with others, use this command.
 \begin{verbatim}
-\RecordProperties{mylabel}
-                 {chapter,page,currentlabel}
+  \RecordProperties{mylabel}
+                   {chapter,page,label}
 \end{verbatim}
-To reference the value the \cs{RefProperty} command is provided which takes 
-two arguments, the label and the property: 
+%%FMi added:
+This records the current values for the properties \texttt{chapter},
+\texttt{page}, and \texttt{label}, using \texttt{mylabel} as a label
+for the record.
+
+%%FMi changed: again following Lamport terminology
+To reference (i.e., to retrieve)
+this recorded value for use in a cross-reference to this chapter, use
+the \cs{RefProperty} command with two arguments: the label and the property.
```

Comment:
add
the label, or \emph{key},

Reply:
Also, for extra clarity, start this as follows (note that only one "to"  is needed 😀 

To \emph{reference} (i.e., retrieve) . . .